src/snowflake/snowpark/_internal/analyzer/select_statement.py

```diff
@@ -181,6 +181,27 @@ def __setitem__(self, col_name: str, col_state: ColumnState) -> None:
                 self.has_new_columns = True
 
 
+def _deepcopy_selectable_fields(
+    from_selectable: "Selectable", to_selectable: "Selectable"
+) -> None:
+    """
+    Make a deep copy of the fields from the from_selectable to the to_selectable
+    """
+    to_selectable.pre_actions = deepcopy(from_selectable.pre_actions)
+    to_selectable.post_actions = deepcopy(from_selectable.post_actions)
+    to_selectable.flatten_disabled = from_selectable.flatten_disabled
+    to_selectable._column_states = deepcopy(from_selectable._column_states)
```

Comment:
do we test if we trigger a `get_attributes` call here?

Reply:
With integ test, i don't see an easy way to see if the describing call is triggered because the describing call doesn't show up in the query history. However, I have manually verified that the describing call is not triggered during the deepcopy process